src/main/java/com/example/diplomnabe/Controllers/RecipeController.java

```diff
@@ -0,0 +1,56 @@
+package com.example.diplomnabe.Controllers;
+import com.example.diplomnabe.Classes.Recipe;
+import com.example.diplomnabe.Classes.User;
+import com.example.diplomnabe.DTO.RecipeDTO;
+import com.example.diplomnabe.DTO.UserDTO;
+import com.example.diplomnabe.Repositories.UserRepository;
+import com.example.diplomnabe.Services.RecipeService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping(path = "api/version1/recipe")
+public class RecipeController
+{
+    private final RecipeService recipeService;
+
+    @Autowired
+    public RecipeController(RecipeService recipeService)
+    {
+        this.recipeService = recipeService;
+    }
+
+    @GetMapping
+    public List<RecipeDTO> getRecipes()
+    {
+        //return recipeService.getRecipes();
+        List<RecipeDTO> list_of_DTO = new ArrayList<>();
```

Comment:
Make sure that all logic is moved to the service including. The mapping of the dto should be in a mapper which you can reuse in your server.

Reply:
The controller should contain code that is handling the request response. Handling errors or calling service function that will handle the logic.